libraries/AP_SmartRTL/AP_SmartRTL.cpp

```diff
@@ -824,11 +824,11 @@ bool AP_SmartRTL::add_loop(uint16_t start_index, uint16_t end_index, const Vecto
 *  This does not matter for the path cleanup algorithm because the pruning will still occur fine between the first
 *  parallel segment and a segment which is directly before or after the second segment.
 */
-AP_SmartRTL::dist_point AP_SmartRTL::segment_segment_dist(const Vector3f &p1, const Vector3f &p2, const Vector3f &p3, const Vector3f &p4)
+AP_SmartRTL::dist_point AP_SmartRTL::segment_segment_dist(const Vector3p &p1, const Vector3p &p2, const Vector3p &p3, const Vector3p &p4)
 {
-    const Vector3f line1 = p2-p1;
-    const Vector3f line2 = p4-p3;
-    const Vector3f line_start_diff = p1-p3; // from the beginning of the second line to the beginning of the first line
+    const Vector3f line1 = (p2 - p1).tofloat();
```

Comment:
I don't think this small optimisation is worth the risk.  the subtractions are done as doubles and then we do a conversion to float so I don't think there is any savings in compute (probably a slight cost).